### PR TITLE
build-sys: fix checking Windows platform with _WIN32 macro

### PR DESCRIPTION
**Pull request by Biswa96**
This replaces WIN32 with _WIN32 macro to check Windows platform.
    The correct one is _WIN32 as documented in the official documentation.
    https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros

### CODE REVIEW

**Comment by masatake** [comment]
I don't know Windows well. However, our own `WIN32` is defined in mk_mingw.mak and mk_mvc.mak.
So I guess our WIN32 has nothing to do with `_WIN32`. Just replacing it may cause trouble.

**Comment by leleliu008** [comment]
@Biswa96 `WIN32` is a user defined macro https://github.com/universal-ctags/ctags/blob/master/mk_mingw.mak#L10, `_WIN32` is a predefined macro which is defined by MSVC or MINGW. what problem have you encountered?

**Comment by Biswa96** [comment]
The issue was seen in geany project, here https://github.com/geany/geany/pull/3660

> `WIN32` is a user defined macro

It is not required at all and contradicts with official and well-known `_WIN32` macro.

File: Units/parser-lex.r/simple-lex.d/input.l [hunk 1]
```diff
@@ -123,7 +123,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifdef WIN32
+#ifdef _WIN32
```
Comment (masatake) [inline Units/parser-lex.r/simple-lex.d/input.l:126]: This is a part of test input. Please don't touch this.

**Comment by masatake** [comment]
Can we remove -DWIN32 from *.mak files?

File: Units/parser-lex.r/simple-lex.d/input.l (hunk 1 above)
Comment (Biswa96) [reply Units/parser-lex.r/simple-lex.d/input.l:126]: Would not the test fail if it is kept unchanged?

**Comment by Biswa96** [comment]
> Can we remove -DWIN32 from *.mak files?

Yes, those can be removed also. I did not notice them at first.

File: Units/parser-lex.r/simple-lex.d/input.l (hunk 1 above)
Comment (masatake) [reply Units/parser-lex.r/simple-lex.d/input.l:126]: No. But this is a part of test input. Please don't touch this.
This change makes your original intent of this pull request unclear.

**Comment by masatake** [comment]
Could you refer to https://github.com/geany/geany/pull/3660 in the commit log?

**Comment by Biswa96** [comment]
I have added the geany pull request in commit message.

File: Units/parser-lex.r/simple-lex.d/input.l (hunk 1 above)
Comment (Biswa96) [reply Units/parser-lex.r/simple-lex.d/input.l]: I have removed that change from the commit.

**Comment by Biswa96** [comment]
Here are some example of the compiler and linker error in geany which are fixed by this change.

```
../ctags/gnu_regex/regex_internal.h:424:11: fatal error: alloca.h: No such file or directory
  424 | # include <alloca.h>
      |           ^~~~~~~~~~

../ctags/main/routines.c:463:21: warning: implicit declaration of function 'lstat'; did you mean 'wstat'? [-Wimplicit-function-declaration]
  463 |                 if (lstat (file.name, &status) != 0)
      |                     ^~~~~
      |                     wstat

ld.exe: libctags.a.p/ctags_main_routines.c.obj:routines.c:(.text+0x7cc): undefined reference to `lstat'
```

**Review by masatake — APPROVED** [review]

**Comment by masatake** [comment]
Can I ask you to change the prefix of the header of the commit log from "build.main,lex" to "build-sys" ?

```
build-sys: fix checking Windows platform with _WIN32 macro
```

**Review by k-takata — APPROVED** [review]

**Pull request merged**